### ISSUE TITLE
Limburg.net not updating from a online source

### ISSUE THREAD

**Issue by Geo-Ron** [ToonSoftwareCollective/wastecollection#6]
Version 4.60 has it removed.

I will try to re-implement it.

Home-Assistant has it going on in [Afvalbeheer](https://github.com/pippyn/Home-Assistant-Sensor-Afvalbeheer/blob/e3f0a983c926497fed1f2337d24afb7618d1b1f9/custom_components/afvalbeheer/sensor.py#L775)

Will try to work this out.

**Comment by ToonzDev** [comment]
Will have a look and copy with pride :-)

**Comment by Geo-Ron** [comment]
In powershell it would be this:

```powershell
$main_url = "https://limburg.net/api-proxy/public"
$City = "PICKONE"
$Streetname = "PICKONE"
$number = 0 #Choose a number
$number_addition = ''
$currDate = Get-Date

$URI = $main_url + "/afval-kalender/gemeenten/search?query={0}" -f $City
$CityResponse = Invoke-RestMethod -Uri $URI

$URI2 = $main_url + "/afval-kalender/gemeente/{0}/straten/search?query={1}" -f $CityResponse.nisCode, $Streetname

$StreetResponse = Invoke-RestMethod -URI $URI2

$URI3 = $main_url + "/kalender/{0}/{1}-{2}?straatNummer={3}&huisNummer={4}&toevoeging={5}" -f $CityResponse.nisCode, $currDate.Year, $currDate.Month, $StreetResponse.nummer, $number, $number_addition

$Events = Invoke-RestMethod -Uri $URI3

# $Legenda = $Events.activiteitenLegende

$Ophalingen = $Events.events

```

**Comment by ToonzDev** [comment]
Hi this will work indeed.
I need to make changes to the config screen to have you enter (parts of) the city name and your streetname.
Data is provided per month so I will retrieve the current and next month and combine the data. Should be enough I guess.
Otherwise relatively easy to implement but quite a change compared to any other waste provider.
Stay tuned.

**Comment by ToonzDev** [comment]
Done most of the work already: configscreen allows you to enter (part of) the city name and street name. The first city and the first street returned are used (configscreen values updated accordingly with the full cityname and streetname). Just need to extract now the dates and waste types from the json. Hopefully done this evening.

**Comment by ToonzDev** [comment]
done, implemented in 4.61 (see more info at the domoticaforum.eu shortly how to use

**Comment by ToonzDev** [comment]
done, implemented in 4.62 (see more info at the domoticaforum.eu shortly how to use)

https://www.domoticaforum.eu/viewtopic.php?f=99&t=11669&p=97037#p97037

**Comment by Geo-Ron** [comment]
Great work!
